sklearn/cross_validation.py

```diff
@@ -433,13 +443,9 @@ def __init__(self, labels, n_folds=3, shuffle=False, random_state=None):
 
         self.idxs = label_to_fold[labels]
```

Comment:
This has a float dtype. It should be initialized with `label_to_fold = np.zeros(n_labels, dtype=np.intp))` instead (at line 433).